src/core/classes/contract-publisher.ts

```diff
@@ -284,14 +284,20 @@ export class ContractPublisher extends RPCConnectionHandler {
         }
       }
       if (p === "bytes32") {
-        return ethers.utils.formatBytes32String(
-          constructorParamValues[index].toString(),
+        invariant(
+          ethers.utils.isHexString(constructorParamValues[index]),
+          `Could not parse bytes32 value. Expected valid hex string but got "${constructorParamValues[index]}".`,
         );
+        if (ethers.utils.isHexString(constructorParamValues[index])) {
```

Comment:
this seems redundant because of the `invariant` function before it. if there's a type issue probably better to just use `constructorParamValues[index] as string`

Reply:
woops - left it on accident. thank you for catching it :P